packages/flutter/lib/src/rendering/flow.dart

```diff
@@ -0,0 +1,345 @@
+  /// containers coordinate system with the given transform. The origin of the
+  /// parent's coordinate system is the upper left corner of the parent, with
+  /// x increasing rightward and y increasing downward.
+  ///
+  /// The container will clip the children to its bounds.
+  void paintChild(int i, Matrix4 transform);
+}
+
+/// A delegate that controls the appearance of a flow layout.
+///
+/// Flow layouts are optimized for moving children around the screen using
+/// transformation matrices. For optimal performance, construct the
+/// [FlowDelegate] with an [Animation] that ticks whenever the delegate wishes
+/// to change the transformation matrices for the children and avoid rebuilding
+/// the [Flow] widget itself every animation frame.
+///
+/// See also:
+///
+///  * [Flow]
+///  * [RenderFlow]
+abstract class FlowDelegate {
+  /// The flow will repaint whenever the [repaint] animation ticks.
+  const FlowDelegate({ Animation<Object> repaint }) : _repaint = repaint;
+
+  final Animation<Object> _repaint;
+
+  /// Override to control the size of the container for the children.
+  ///
+  /// By default, the flow will be as large as possible. If this function
+  /// return a size that does not respect the given constraints, the size will
+  /// be adjusted to be as close to the returned size as possible while still
+  /// respecting the constraints.
+  ///
+  /// If this function depends on information other than the given constraints,
+  /// override [shouldRelayout] to indicate when when the container should
+  /// relayout.
+  Size getSize(BoxConstraints constraints) => constraints.biggest;
+
+  /// Override to control the layout constraints given to each child.
+  ///
+  /// By default, the children will receive the given constraints, which are the
+  /// constraints the constraints used to size the container. The children need
+  /// not respect the given constraints, but they are required to respect the
+  /// returned constraints.
+  ///
+  /// If this function depends on information other than the given constraints,
+  /// override [shouldRelayout] to indicate when when the container should
+  /// relayout.
+  BoxConstraints getConstraintsForChild(int i, BoxConstraints constraints) => constraints;
+
+  /// Override to paint the children of the flow.
+  ///
+  /// Children can be painted in any order, but each child can be painted at
+  /// most once. Although the container clips the children to its own bounds, it
+  /// is more efficient to skip painting a child altogether rather than having
+  /// it paint entirely outside the container's clip.
+  ///
+  /// To paint a child, call [FlowPaintingContext.paintChild] on the given
+  /// [context]. The given context is valid only within the scope of this
+  /// function call and contains information (such as the size of the container)
+  /// that is useful for picking transformation matrices for the children.
```

Comment:
since you mention shouldrelayout for all the other functions, maybe mention shouldrepaint here?